torch/csrc/cudnn/Conv.cpp

```diff
@@ -225,6 +225,14 @@ void cudnn_convolution_forward(
   size_t workspaceSize;
   CHECK(cudnnGetConvolutionForwardWorkspaceSize(handle, idesc.desc, wdesc.desc,
       cdesc.desc, odesc.desc, fwdAlg, &workspaceSize));
+  size_t memFree;
+  size_t memTotal;
+  cudaMemGetInfo(&memFree, &memTotal);
```

Comment:
@fmassa I think that the problem is that the workspace size returned might be larger than can be satisfied by the caching allocator. In this case `Workspace` will call `THCudaMalloc` and that will give you OOM. But I haven't verified that, it's just a guess.